### PR TITLE
feat: add the cppcheck runner

### PR DESCRIPTION
**Pull request by swarnim-deepsource**
Task:
* [x] Add the rust code for the cppcheck deepsource runner.
* [x] Add the docker file.
* [x] Update readme and automate pipeline for image upload.

Fixes lae-7799

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        files_iter,
+        cppcheck_errors_path
+            .to_str()
+            .unwrap_or("/toolbox/cppcheck_error.xml"),
+    );
+
+    let mut issue_occurrences = vec![];
+    if let Some(cppcheck_results) = std::fs::read_to_string(cppcheck_errors_path)
+        .ok()
+        .map(|src| quick_xml::de::from_str::<cpkerrs::Results>(&src).unwrap())
```
Comment (raghav-deepsource) [inline src/main.rs]: maybe do some kind of map ok here

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        log::error!("{err}");
+        "{}".to_string()
+    });
+
+    use std::io::Write;
+    write!(
+        std::fs::File::create(toolbox_directory.join("cppcheck_result.json"))?,
+        "{}",
+        src
+    )?;
```
Comment (raghav-deepsource) [inline src/main.rs]: try using std::fs::write here, looks suitable.
https://doc.rust-lang.org/std/fs/fn.write.html

File: OWNERS [hunk 3]
```diff
@@ -0,0 +1,10 @@
+approvers:
```
Comment (prajwal-deepsource) [inline OWNERS]: Is this file required? We don't use Lighthouse anymore, I think it makes more sense to have this as a `.github/CODEOWNERS` file instead

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# Get Sentry
+ARG SENTRY_TAG=0.5.2
+RUN mkdir /sentry-sdk \
+&& cd /sentry-sdk \
+&& wget --no-verbose "https://github.com/getsentry/sentry-native/releases/download/${SENTRY_TAG}/sentry-native.zip" \
+&& unzip sentry-native.zip \
+&& cmake -B ./build \
+&& cmake --build ./build --parallel \
+&& cmake --install ./build --prefix "${SENTRY_INSTALL_DIR}"
+# -----------------------------------------------------------
```
Comment (deepak-deepsource) [inline Dockerfile:50]: Let's add the following for spdlog.

```
RUN git clone https://github.com/gabime/spdlog.git \
  && cd spdlog \
  && cmake -B build \
  && cmake --build ./build --parallel \
  && cd build && make install
```

File: OWNERS (hunk 3 above)
Comment (deepak-deepsource) [reply OWNERS]: @swarnim-deepsource Can you take care of this?

**Review by deepak-deepsource — CHANGES_REQUESTED** [review]

**Review by deepak-deepsource — APPROVED** [review]
